network/eventManager.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2017 The Alvalor Authors
+//
+// This file is part of Alvalor.
+//
+// Alvalor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Alvalor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with Alvalor.  If not, see <http://www.gnu.org/licenses/>.
+
+package network
+
+import (
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+type eventManager interface {
+	Disconnected(addr string)
+	Connected(addr string)
+	Received(addr string, msg interface{})
+}
+
+type simpleEventManager struct {
+	subscriber chan interface{}
```

Comment:
I just noticed that a nice side effect of putting the subscriber channel here is the fact that we can easily change the channel without having access to all the processor goroutines. This is nice in case we want to change the semantics of the `Subscribe` API, for example to allow the consumer to provide the channel so he can decide the size of the buffer himself. It was something I was thinking about and now it's already solved :)

Reply:
Definitely nice side effect :)